syncopy/io/mne_conv.py

```diff
@@ -0,0 +1,134 @@
+
+    Returns
+    -------
+    epochs : `mne.EpochsArray` instance
+    """
+    try:
+        import mne
+    except ImportError:
+        raise ImportError("MNE Python not installed, but package 'mne' is required for this function.")
+    assert type(tldata) == spy.AnalogData or type(tldata) == spy.TimeLockData, "Invalid input: tldata must be of type AnalogData or TimeLockData."
+    if type(tldata) == spy.AnalogData:
+        assert tldata.is_time_locked == True, "Invalid input: tldata must be time-locked."
+    info = mne.io.meas_info.create_info(list(tldata.channel), tldata.samplerate, ch_types='misc')
+
+    # for MNE, the data needs to have shape (n_epochs, n_channels, n_times) but our
+    # TimeLockData has shape (n_times, n_channels) with trials concatenated along the time axis
+    num_trials = len(tldata.trials)
+    num_channels = len(tldata.channel)
+    trial_len = tldata.trials[0].shape[0]  # Known to be identical for all trials to due to is_time_locked() check
+    data_with_trial_axis = np.zeros((num_trials, num_channels, trial_len), dtype=tldata.data.dtype)
+    for trial_idx in range(len(tldata.trials)):
+        data_with_trial_axis[trial_idx,:,:] = tldata.trials[trial_idx].T
+
+    ea = mne.EpochsArray(data_with_trial_axis, info)
+    return ea
+
+def mne_epochs_to_tldata(ea):
+    """
+    Convert MNE EpochsArray to Syncopy TimeLockData.
+
+    This function requires MNE Python (package 'mne') and will raise an `ImportError` if it is not installed.
+
+    Parameters
+    ----------
+    ea : `mne.EpochsArray` instance
+
+    Returns
+    -------
+    tldata : `syncopy.TimeLockData` instance
+    """
+    try:
+        import mne
+    except ImportError:
+        raise ImportError("MNE Python not installed, but package 'mne' is required for this function.")
+    assert type(ea) == mne.EpochsArray, "Invalid input: ea must be of type mne.EpochsArray."
+    # ed.data has shape (n_epochs, n_channels, n_times), convert to spy_data with shape (n_times, n_channels) with epochs concatenated along the time axis
+    n_epochs = ea.get_data().shape[0]
+    n_channels = ea.get_data().shape[1]
+    n_times = ea.get_data().shape[2]
+    spy_data = np.zeros((n_times * n_epochs, n_channels), dtype=np.float32)
+
+    for chan_idx in range(n_channels):
+        spy_data[:, chan_idx] = ea.get_data()[:,chan_idx,:].flatten()
+
+    tldata = spy.AnalogData(data=spy_data, samplerate=ea.info['sfreq'], channel=ea.ch_names)
+
+    nSamples = n_times
+    trldef = np.vstack([np.arange(0, nSamples * n_epochs, nSamples),
+                        np.arange(0, nSamples * n_epochs, nSamples) + nSamples,
+                        np.ones(n_epochs) * -1]).T
```

Comment:
so here all offsets are set to `-1`? This determines the `0` of the time-axis, not sure how this is encoded in MNE actually..

Reply:
I think because the imported data is epoched already, we want our trials to span the entire input data. Should this be `np.zeros(n_epochs)` then instad of `np.ones(n_epochs) * -1])`?

Reply:
I think `offset = -ea.tmin * samplerate` would do the trick, then just  `np.ones(n_epochs) * offset` as last column in the trialdef.